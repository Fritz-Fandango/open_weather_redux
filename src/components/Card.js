import React, { Component } from 'react';
import '../styles/Card.scss';


function formatDate (timestamp) {
    const event = new Date(timestamp);

    return event.toLocaleDateString('en-US');
}

function formatTime (timestamp) {
    const event = new Date(timestamp);

    return event.toLocaleTimeString('en-US');
}

class Card extends Component {
    render() {
        return (
            <div className='card'>
                <img
                    src={`http://openweathermap.org/img/wn/${this.props.icon}@2x.png`}
                    alt='Avatar'
                    style={{width: '100%' }}
                />
                <div className='container'>
                    <p>Date: {formatDate(this.props.dateTime)}</p>
                    <p>Min temp: {this.props.minTemp}&deg;F</p>
                    <p>Max temp: {this.props.maxTemp}&deg;F</p>
                    <p>Humidty: {this.props.humidity}%</p>
                    <p>Sunrise: {formatTime(this.props.sunrise)}</p>
                    <p>Sunset: {formatTime(this.props.sunset)}</p>
                </div>
            </div>
        )
    }
}

export default Card;
