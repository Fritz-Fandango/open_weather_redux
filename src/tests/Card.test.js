import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';

import {
    formatDate,
    formatTime,
} from '../components/Card';

import Card from '../components/Card';

configure({adapter: new Adapter()});

describe('Testing date formatting', () => {
    const timestamp = 1618308000; // 1/19/1970 10:31:48 AM

    it('should return a valid date string', () => {
        const formattedDate = formatDate(timestamp);

        expect(formattedDate).toBe('1/19/1970');
    });

    it('should return a valid time string', () => {
        const formattedDate = formatTime(timestamp);

        expect(formattedDate).toBe('10:31:48 AM');
    });

});

describe('Testing Card component', () => {
    const propsObj = {
        key: 1618308000,
        dateTime: 1618308000,
        humidity: 15,
        icon: '10d',
        minTemp: 275.09,
        maxTemp: 279.79,
        sunrise: 1618282134,
        sunset: 1618333901
    };

    it('renders Card', () => {
        const wrapper = <Card props={propsObj} />;

        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
