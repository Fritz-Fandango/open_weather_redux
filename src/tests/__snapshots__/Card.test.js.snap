// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card component renders Card 1`] = `
<div
  className="card"
>
  <img
    alt="Avatar"
    src="http://openweathermap.org/img/wn/undefined@2x.png"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="container"
  >
    <p>
      Date: 
      Invalid Date
    </p>
    <p>
      Min temp: 
      °F
    </p>
    <p>
      Max temp: 
      °F
    </p>
    <p>
      Humidty: 
      %
    </p>
    <p>
      Sunrise: 
      Invalid Date
    </p>
    <p>
      Sunset: 
      Invalid Date
    </p>
  </div>
</div>
`;
